analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CreatingHashAlgorithms.cs

```diff
@@ -1,77 +1,104 @@
-﻿namespace NS
-{
-    using System.Security.Cryptography;
+﻿using System.Security.Cryptography;
 
-    public class TestClass
+namespace Tests.Diagnostics
+{
+    public class InsecureHashAlgorithm
     {
-        // RSPEC 4790: https://jira.sonarsource.com/browse/RSPEC-4790
-        public void ComputeHash()
+        public void Hash(byte[] temp)
         {
-            // Review all instantiations of classes that inherit from HashAlgorithm, for example:
-            HashAlgorithm hashAlgo = HashAlgorithm.Create();
-//                                   ^^^^^^^^^^^^^^^^^^^^^^    {{Make sure this weak hash algorithm is not used in a sensitive context here.}}
-            HashAlgorithm hashAlgo2 = HashAlgorithm.Create("SHA1");
-//                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^    {{Make sure this weak hash algorithm is not used in a sensitive context here.}}
+            using (var sha1 = new SHA1Managed()) // Noncompliant {{Make sure this weak hash algorithm is not used in a sensitive context here.}}
+//                            ^^^^^^^^^^^^^^^^^
+            {
+            }
+            using (var sha1 = new SHA1CryptoServiceProvider()) // Noncompliant
+            {
+            }
 
-            SHA1 sha = new SHA1CryptoServiceProvider();
-//                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^    {{Make sure this weak hash algorithm is not used in a sensitive context here.}}
+            using var sha1csharp8 = new SHA1Managed(); // Noncompliant
 
-            MD5 md5 = new MD5CryptoServiceProvider();
-//                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^    {{Make sure this weak hash algorithm is not used in a sensitive context here.}}
-
-            // ...
+            using (var sha256 = new SHA256Managed())
+            {
+            }
+            using (var sha256 = (HashAlgorithm)CryptoConfig.CreateFromName("SHA256Managed"))
+            {
+            }
+            using (var sha256 = HashAlgorithm.Create("SHA256Managed"))
+            {
+            }
         }
 
-        public void AdditionalTests(SHA1CryptoServiceProvider sha1)
+        public void Md5Calls()
         {
-            var myHash = new MyHashAlgorithm();     // Noncompliant
```

Comment:
This is gone now, because it was meanwhile handled by S2257?

Reply:
No, the reason is that we want to mirror the functionality from S2070 which does not check for custom inheritance of `HashAlgorithm`